x-pack/plugins/lens/public/datasources/text_based/utils.test.ts

```diff
@@ -66,6 +66,22 @@ describe('Text based languages utils', () => {
 
       expect(indexPattern).toBe('');
     });
+
+    it('should return the index pattern for es|ql query', () => {
+      const indexPattern = getIndexPatternFromTextBasedQuery({
+        sql: 'from foo | keep bytes, memory ',
```

Comment:
```suggestion
        esql: 'from foo | keep bytes, memory ',
```
Should this be `esql`?